src/kernels/jupyter/commands/serverSelector.ts

```diff
@@ -48,7 +48,7 @@ export class JupyterServerSelectorCommand implements IDisposable {
             // Find one that is the default for this remote
             if (notebook) {
                 // Recompute the preferred controller
-                await this.controllerManager.computePreferredNotebookController(notebook);
+                await this.controllerManager.initializePreferredNotebookController(notebook);
```

Comment:
This was not only computing the preferred controller, it was also setting it using the VS Code api.
I've renamed this to be more specific.
The method `computePreferredNotebookController` is now used to return the preferred controller without changing it.